diff --git a/src/cpu/dcn_v2_cpu.cpp b/src/cpu/dcn_v2_cpu.cpp
index e20b56c..4aa7b4f 100644
--- a/src/cpu/dcn_v2_cpu.cpp
+++ b/src/cpu/dcn_v2_cpu.cpp
@@ -204,35 +204,35 @@ std::vector<at::Tensor> dcn_v2_cpu_backward(const at::Tensor &input,
         columns = at::matmul(weight_flat, grad_output_n_flat);
 
         // gradient w.r.t. input coordinate data
-        modulated_deformable_col2im_coord_cpu(columns.data<scalar_t>(),
-                                               input_n.data<scalar_t>(),
-                                               offset_n.data<scalar_t>(),
-                                               mask_n.data<scalar_t>(),
+        modulated_deformable_col2im_coord_cpu(columns.data_ptr<scalar_t>(),
+                                               input_n.data_ptr<scalar_t>(),
+                                               offset_n.data_ptr<scalar_t>(),
+                                               mask_n.data_ptr<scalar_t>(),
                                                1, channels, height, width,
                                                height_out, width_out, kernel_h, kernel_w,
                                                pad_h, pad_w, stride_h, stride_w,
                                                dilation_h, dilation_w, deformable_group,
-                                               grad_offset_n.data<scalar_t>(),
-                                               grad_mask_n.data<scalar_t>());
+                                               grad_offset_n.data_ptr<scalar_t>(),
+                                               grad_mask_n.data_ptr<scalar_t>());
         // gradient w.r.t. input data
-        modulated_deformable_col2im_cpu(columns.data<scalar_t>(),
-                                         offset_n.data<scalar_t>(),
-                                         mask_n.data<scalar_t>(),
+        modulated_deformable_col2im_cpu(columns.data_ptr<scalar_t>(),
+                                         offset_n.data_ptr<scalar_t>(),
+                                         mask_n.data_ptr<scalar_t>(),
                                          1, channels, height, width,
                                          height_out, width_out, kernel_h, kernel_w,
                                          pad_h, pad_w, stride_h, stride_w,
                                          dilation_h, dilation_w, deformable_group,
-                                         grad_input_n.data<scalar_t>());
+                                         grad_input_n.data_ptr<scalar_t>());
 
         // gradient w.r.t. weight, dWeight should accumulate across the batch and group
-        modulated_deformable_im2col_cpu(input_n.data<scalar_t>(),
-                                         offset_n.data<scalar_t>(),
-                                         mask_n.data<scalar_t>(),
+        modulated_deformable_im2col_cpu(input_n.data_ptr<scalar_t>(),
+                                         offset_n.data_ptr<scalar_t>(),
+                                         mask_n.data_ptr<scalar_t>(),
                                          1, channels, height, width,
                                          height_out, width_out, kernel_h, kernel_w,
                                          pad_h, pad_w, stride_h, stride_w,
                                          dilation_h, dilation_w, deformable_group,
-                                         columns.data<scalar_t>());
+                                         columns.data_ptr<scalar_t>());
 
         // long m_ = channels_out;
         // long n_ = channels * kernel_h * kernel_w;
diff --git a/src/cpu/dcn_v2_psroi_pooling_cpu.cpp b/src/cpu/dcn_v2_psroi_pooling_cpu.cpp
index cc5adc0..8abba43 100644
--- a/src/cpu/dcn_v2_psroi_pooling_cpu.cpp
+++ b/src/cpu/dcn_v2_psroi_pooling_cpu.cpp
@@ -318,17 +318,17 @@ dcn_v2_psroi_pooling_cpu_forward(const at::Tensor &input,
   /*dim3 grid(std::min(at::ceil_div(out_size, 512L), 4096L));
   dim3 block(512);*/
 
-  AT_DISPATCH_FLOATING_TYPES(input.type(), "dcn_v2_psroi_pooling_cpu_forward", [&] {
+  AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "dcn_v2_psroi_pooling_cpu_forward", [&] {
     DeformablePSROIPoolForwardKernelCpu<scalar_t>(
         out_size,
-        input.contiguous().data<scalar_t>(),
+        input.contiguous().data_ptr<scalar_t>(),
         spatial_scale,
         channels,
         height, width,
         pooled_height,
         pooled_width,
-        bbox.contiguous().data<scalar_t>(),
-        trans.contiguous().data<scalar_t>(),
+        bbox.contiguous().data_ptr<scalar_t>(),
+        trans.contiguous().data_ptr<scalar_t>(),
         no_trans,
         trans_std,
         sample_per_part,
@@ -337,8 +337,8 @@ dcn_v2_psroi_pooling_cpu_forward(const at::Tensor &input,
         part_size,
         num_classes,
         channels_each_class,
-        out.data<scalar_t>(),
-        top_count.data<scalar_t>());
+        out.data_ptr<scalar_t>(),
+        top_count.data_ptr<scalar_t>());
   });
   //C10_CUDA_CHECK(cudaGetLastError());
   return std::make_tuple(out, top_count);
@@ -392,11 +392,11 @@ dcn_v2_psroi_pooling_cpu_backward(const at::Tensor &out_grad,
   dim3 block(512);
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();*/
 
-  AT_DISPATCH_FLOATING_TYPES(out_grad.type(), "dcn_v2_psroi_pooling_cpu_backward", [&] {
+  AT_DISPATCH_FLOATING_TYPES(out_grad.scalar_type(), "dcn_v2_psroi_pooling_cpu_backward", [&] {
     DeformablePSROIPoolBackwardAccKernelCpu<scalar_t>(
         out_size,
-        out_grad.contiguous().data<scalar_t>(),
-        top_count.contiguous().data<scalar_t>(),
+        out_grad.contiguous().data_ptr<scalar_t>(),
+        top_count.contiguous().data_ptr<scalar_t>(),
         num_bbox,
         spatial_scale,
         channels,
@@ -405,11 +405,11 @@ dcn_v2_psroi_pooling_cpu_backward(const at::Tensor &out_grad,
         pooled_height,
         pooled_width,
         output_dim,
-        input_grad.contiguous().data<scalar_t>(),
-        trans_grad.contiguous().data<scalar_t>(),
-        input.contiguous().data<scalar_t>(),
-        bbox.contiguous().data<scalar_t>(),
-        trans.contiguous().data<scalar_t>(),
+        input_grad.contiguous().data_ptr<scalar_t>(),
+        trans_grad.contiguous().data_ptr<scalar_t>(),
+        input.contiguous().data_ptr<scalar_t>(),
+        bbox.contiguous().data_ptr<scalar_t>(),
+        trans.contiguous().data_ptr<scalar_t>(),
         no_trans,
         trans_std,
         sample_per_part,
diff --git a/src/dcn_v2.h b/src/dcn_v2.h
index de670bf..cffa4e9 100644
--- a/src/dcn_v2.h
+++ b/src/dcn_v2.h
@@ -22,7 +22,7 @@ dcn_v2_forward(const at::Tensor &input,
                const int dilation_w,
                const int deformable_group)
 {
-    if (input.type().is_cuda())
+    if (input.is_cuda())
     {
 #ifdef WITH_CUDA
         return dcn_v2_cuda_forward(input, weight, bias, offset, mask,
@@ -58,7 +58,7 @@ dcn_v2_backward(const at::Tensor &input,
                 int dilation_h, int dilation_w,
                 int deformable_group)
 {
-    if (input.type().is_cuda())
+    if (input.is_cuda())
     {
 #ifdef WITH_CUDA
         return dcn_v2_cuda_backward(input,
@@ -104,7 +104,7 @@ dcn_v2_psroi_pooling_forward(const at::Tensor &input,
                              const int sample_per_part,
                              const float trans_std)
 {
-    if (input.type().is_cuda())
+    if (input.is_cuda())
     {
 #ifdef WITH_CUDA
         return dcn_v2_psroi_pooling_cuda_forward(input,
@@ -152,7 +152,7 @@ dcn_v2_psroi_pooling_backward(const at::Tensor &out_grad,
                               const int sample_per_part,
                               const float trans_std)
 {
-    if (input.type().is_cuda())
+    if (input.is_cuda())
     {
 #ifdef WITH_CUDA
         return dcn_v2_psroi_pooling_cuda_backward(out_grad,
